File: packages/backend/app/schemas/build_plan.py
from pydantic import BaseModel


class BuildPlanRequest(BaseModel):
    project_id: int | None = None
    description: str
    room_id: int | None = None
    scope: str = ""
    skill_level: str = "beginner"


class BuildStepSchema(BaseModel):
    order: int
    title: str
    description: str
    tools: list[str]
    materials: list[dict]
    safety_warnings: list[str]
    estimated_mins: int | None
    image_url: str | None = None

    class Config:
        from_attributes = True


class BuildPlanResponse(BaseModel):
    id: int
    project_id: int
    title: str
    difficulty: str
    estimated_hours: float | None
    total_cost_est: float | None
    safety_level: str | None
    steps: list[BuildStepSchema]
    created_at: str

    class Config:
        from_attributes = True

File: packages/shared/src/types/build-plan.ts
export interface BuildPlan {
  id: number;
  projectId: number;
  title: string;
  difficulty: 'makkelijk' | 'gemiddeld' | 'moeilijk';
  estimatedHours: number | null;
  totalCostEstimate: number | null;
  safetyLevel: 'laag' | 'middel' | 'hoog';
  steps: BuildStep[];
  createdAt: string;
}

export interface BuildStep {
  id: number;
  order: number;
  title: string;
  description: string;
  tools: string[];
  materials: MaterialItem[];
  safetyWarnings: string[];
  estimatedMinutes: number | null;
  imageUrl: string | null;
  videoUrl: string | null;
  completed: boolean;
}

export interface MaterialItem {
  productId: number | null;
  name: string;
  quantity: number;
  unit: string;
  estimatedPrice: number | null;
}

export interface BuildPlanRequest {
  projectId: number;
  description: string;
  roomId: number | null;
  scope: string;
  skillLevel: 'beginner' | 'gemiddeld' | 'ervaren';
}
